src/main/java/edu/cornell/kfs/sys/dataaccess/impl/PreferencesDaoDynamoDB.java

```diff
@@ -193,11 +195,21 @@ public void saveUserPreferences(String principalName, Map<String, Object> prefer
 
     protected Table retrieveTable(String tableName) {
         DynamoDB dynamoDB = new DynamoDB(documentStoreClient.getClient());
+        if (StringUtils.isNotBlank(tableNamePrefix)) {
+            tableName = tableNamePrefix + tableName;
```

Comment:
It never gets called with a null tableName - constants are used for those. I think if somehow it did end up with an invalid table name because it had prefix + null, I think later on down the line you'd just get an error that the table didn't exist.